### PR TITLE
Reorganize the upload and attach CLI commands.

### PR DESCRIPTION
**Pull request by dlorenc**
This is in preparation for #368.

Signed-off-by: Dan Lorenc <dlorenc@google.com>

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -125,21 +125,21 @@ The signature is passed via the -signature flag.
 It can be a file:
 
 ```shell
-$ cosign upload -signature file.sig dlorenc/demo
+$ cosign attach signature -signature file.sig dlorenc/demo
```
Comment (dekkagaijin) [inline USAGE.md:128]: can drop `-signature` in this case and just have 2 positionals

Comment (dlorenc) [reply USAGE.md:128]: I just went back and forth on this one and decided against it. -signature is long but two positionals is uglier

Comment (dekkagaijin) [reply USAGE.md:128]: /shrug

File: USAGE.md [hunk 2]
```diff
@@ -125,21 +125,21 @@ The signature is passed via the -signature flag.
...
-$ cosign upload -signature Qr883oPOj0dj82PZ0d9mQ2lrdM0lbyLSXUkjt6ejrxtHxwe7bU6Gr27Sysgk1jagf1htO/gvkkg71oJiwWryCQ== dlorenc/demo
+$ cosign attach signature -signature Qr883oPOj0dj82PZ0d9mQ2lrdM0lbyLSXUkjt6ejrxtHxwe7bU6Gr27Sysgk1jagf1htO/gvkkg71oJiwWryCQ== dlorenc/demo
 Pushing signature to: dlorenc/demo:sha256-87ef60f558bad79beea6425a3b28989f01dd417164150ab3baab98dcbf04def.sig
 ```
 
 or, `-` for stdin for chaining from other commands:
 
 ```shell
-$ cosign generate dlorenc/demo | openssl... | cosign upload -signature -- dlorenc/demo
+$ cosign generate dlorenc/demo | openssl... | cosign attach signature -signature -- dlorenc/demo
```
Comment (dekkagaijin) [inline USAGE.md:142]: should just be one `-`?

**Review by dekkagaijin — APPROVED** [review]

File: USAGE.md (hunk 2 above)
Comment (dlorenc) [reply USAGE.md:142]: No I don't think so, the -- indicates to read from STDIN

**Pull request merged**